fixed dead links and added http/https

G-Square opened the pull request.

onebree commented on free-programming-books.md:
```diff
@@ -1082,6 +1087,7 @@ Original Source: [List of freely available programming books](http://web.archive
 * [Scalable and Modular Architecture for CSS](http://smacss.com) - Jonathan Snook
 * [Web Audio API](http://chimera.labs.oreilly.com/books/1234000001552) - Boris Smus
+* [What Is HTML5?](http://shop.oreilly.com/product/0636920021049.do) - Brett McLaughlin
 * [Adaptive Web Design] (http://adaptivewebdesign.info/1st-edition/) - Aaron Gustafson
```
I know you did not submit this link, but can you ABC order it?

G-Square replied:
I think it would be a good idea if someone would write a format validator script (python,ruby,etc) to take care of this before pull request. It would take care of common problems like spaces , `] (` and order all sections accordingly. And any other format problem @vhf would like to add.

onebree commented:
If you are familiar with command-line git, please squash all these commits into one. Every time you hit "save" on Github, it creates 1 commit.

vhf commented:
For this kind of project, I tend to prefer one commit per resource addition / deletion.

But if anyone finds it tedious when adding several resources, one big commit is also fine for me. I don't want to enforce strict commit rules. But if we can avoid squashing, I still prefer one commit per link.

G-Square commented:
I added `http` and `https` to some commits, this was missing in closed pull request #1652 (I deleted my fork). 
And removed some of the links that stopped working for some reason.
Will keep squash in mind, Thanks @onebree.

vhf commented:
I did my best trying to merge this patch, but did not succeed.

    error: patch failed: free-programming-books.md:1520
    error: free-programming-books.md: patch does not apply

vhf commented:
Unfortunately I'm note sure. I could have fought a bit longer with git, and probably would have found a solution. But since @G-Square 's branch does not exist anymore, and git does not want me to apply the patch, I'm a bit lost. I'll reopen and do my best.

vhf commented:
Hmm, unfortunately github does not allow me to reopen the PR. When I hover the "Reopen" button, it says "The repository that submitted this pull reuqest has been deleted". I guess I'll have to go through the changes and apply them manually when it still applies.
